omniqueue/src/queue/consumer.rs

```diff
@@ -72,20 +82,44 @@ impl<C: QueueConsumer> ErasedQueueConsumer for DynConsumerInner<C> {
             Ok(out)
         })
     }
+
+    fn max_messages(&self) -> Option<NonZeroUsize> {
+        self.inner.max_messages()
+    }
 }
 
 impl DynConsumer {
     pub async fn receive(&mut self) -> Result<Delivery> {
         self.0.receive().await
     }
 
+    /// Receive up to `max_messages` from the queue, waiting up to `deadline`
+    /// for more messages to arrive.
+    ///
+    /// Unlike the `receive_all` methods on specific backends, this method
+    /// clamps `max_messages` to what's permissible by the backend, so you don't
+    /// have to know which backend is actually in use as a user of this type.
     pub async fn receive_all(
         &mut self,
         max_messages: usize,
```

Comment:
More curious than anything. Is there a use case for passing `0` for `max_message`? The non-zero types are a bit fiddly for a public API so I wouldn't force that hassle on callers, but it just makes me wonder 🤔

Reply:
Can't think of anything.